telemetry/telemetrybuffer.go

```diff
@@ -116,6 +116,27 @@ func (tb *TelemetryBuffer) StartServer() error {
 				tb.connections = append(tb.connections, conn)
 				tb.mutex.Unlock()
 				go func() {
+					defer func() {
+						var index int
+						var value net.Conn
+						var found bool
+
+						tb.mutex.Lock()
+						defer tb.mutex.Unlock()
+
+						for index, value = range tb.connections {
+							if value == conn {
+								conn.Close()
+								found = true
+								break
+							}
+						}
+
+						if found {
+							tb.connections = remove(tb.connections, index)
+						}
+					}()
+
 					for {
 						reportStr, err := read(conn)
 						if err == nil {
```

Comment:
I suggest inverting this now, we tend to prefer to handle errors first, so we should do
```go
if err != nil {
    return
}
```
then we can drop the else too, so we drop a level of indentation